packages/wallets/src/components/WalletsCarouselContent/WalletsCarouselContent.tsx

```diff
@@ -10,39 +10,51 @@ import './WalletsCarouselContent.scss';
 const WalletsCarouselContent: React.FC = () => {
     const { isLoading: isAuthorizeLoading, switchAccount } = useAuthorize();
     const { isLoading: isCurrencyConfigLoading } = useCurrencyConfig();
+    const { data: walletAccountsList } = useWalletAccountsList();
+    const [isCarouselInitialized, setIsCarouselInitialized] = useState(false);
     const [walletsCarouselEmblaRef, walletsCarouselEmblaApi] = useEmblaCarousel({
         containScroll: false,
         skipSnaps: true,
     });
-    const { data: walletAccountsList } = useWalletAccountsList();
+
+    useEffect(() => {
+        if (walletsCarouselEmblaApi) {
+            setIsCarouselInitialized(true);
+        }
+    }, [walletsCarouselEmblaApi]);
```

Comment:
Due to the asynchronous nature of the useMemo and useEffect hooks. The scrollTo method might be called before the carousel is fully initialized or before the activeWalletIndex is correctly calculated